packages/plugin-coverage/src/lib/nx/coverage-paths.unit.test.ts

```diff
@@ -193,32 +259,90 @@ describe('getCoveragePathForJest', () => {
   it('should fetch Jest coverageDirectory', async () => {
     await expect(
       getCoveragePathForJest(
-        'jest-valid.config.unit.ts',
-        join('packages', 'cli'),
-        'unit-test in core',
+        { jestConfig: 'jest-valid.config.unit.ts' },
+        { name: 'cli', root: join('packages', 'cli') },
+        'unit-test',
       ),
     ).resolves.toBe(join('packages', 'cli', 'coverage', 'core', 'lcov.info'));
   });
 
-  it('should throw when reportsDirectory is not set in Jest config', async () => {
+  it('should throw when coverageDirectory is not set in Jest config', async () => {
     await expect(() =>
       getCoveragePathForJest(
-        'jest-no-dir.config.integration.ts',
-        'packages',
-        'integration-test in core',
+        { jestConfig: 'jest-no-dir.config.integration.ts' },
+        { name: 'core', root: join('packages', 'core') },
+        'integration-test',
       ),
     ).rejects.toThrow(
       /configuration .* does not include coverage path .* Add the path under coverageDirectory/,
     );
   });
 
+  it('should override jest config coverageDirectory from project.json', async () => {
+    await expect(
+      getCoveragePathForJest(
+        {
+          jestConfig: 'jest-valid.config.unit.ts',
+          coverageDirectory: join(
+            '..',
+            '..',
+            'dist',
+            'packages',
+            'cli',
+            'coverage',
+          ),
+        },
+        { name: 'cli', root: join('packages', 'cli') },
+        'unit-test',
+      ),
+    ).resolves.toBe(join('dist', 'packages', 'cli', 'coverage', 'lcov.info'));
+  });
+
   it('should throw when Jest config does not include lcov reporter', async () => {
     await expect(() =>
       getCoveragePathForJest(
-        'jest-no-lcov.config.integration.ts',
-        'packages',
-        'integration-test in core',
+        { jestConfig: 'jest-no-lcov.config.integration.ts' },
+        { name: 'core', root: join('packages', 'core') },
+        'integration-test',
       ),
     ).rejects.toThrow(/configuration .* does not include LCOV report format/);
   });
+
+  it('should not throw if lcov reporter in project.json instead of jest config', async () => {
```

Comment:
📖 Functionality documentation: It would be valuable to include here which value is prioritised in the case of it being provided both in `project.json` and the config.

Reply:
Added a test case, although it seems a bit pointless to me. If lcov is set in both `jest.config.ts` and `project.json`, then obviously it doesn't throw either. How could the answer be anything else?

Reply:
Thought of a more interesting test case just now :thinking: - `jest.config.ts` sets lcov, but `project.json` overrides it with something else. Added a test for it.